### DISCUSSION TITLE
Competing risk data

### DISCUSSION THREAD

**Discussion by Justinewits** [kogalur/randomForestSRC#193] [Show and tell]
Dear Prof,
  Can you please share your R code to show how  you created the competing risk dataset(s) in **your 2014 paper on Random survival forests for competing risks**. Especially where you stated  that "The additive structure of the linear model was changed in our quadratic model” and  "Additional interaction effects were added to the linear model of the form".  I really appreciate your help, thank you!
  
Below is my attempt on the quadratic case.  Please let me know whether I am on the right track. 
 x1 <-rnorm(N, mean=0, sd=1) 
  x2 <-rnorm(N, mean=0, sd=1)
  x3 <-rnorm(N, mean=0, sd=1)
  x4 <-rnorm(N, mean=0, sd=1)
  x5 <-rnorm(N, mean=0, sd=1)
  x6 <-rnorm(N, mean=0, sd=1)
  x7 <- rbinom(N,1,0.5)
  x8 <- rbinom(N,1,0.5)
  x9 <- rbinom(N,1,0.5)
  x10 <- rbinom(N,1,0.5)
  x11 <- rbinom(N,1,0.5)
  x12 <- rbinom(N,1,0.5)
  x13<-x1^2
  x14<-x2^2
  x15<-x3^2
  x16<-x4^2
  x17<-x5^2
  x18<-x6^2
  rate = 0.01
  b1 = log(2)
  b2 = 1.5
  betas1 <- c(b1,-b1,0,0,b1,-b1,b2,-b2,0,0,b2,-b2)
  betas2 <- c(0,0,b1,-b1,b1,-b1,0,0,b2,-b2,b2,-b2)
  betas1Q<-c(b1,-b1,0,0,b1,-b1)
  betas2Q<-c(0,0,b1,-b1,b1,-b1)

  # create the two hazards for the events 
  lambda1<-rate*exp(betas1[1]*x1+betas1[2]*x2+betas1[3]*x3+betas1[4]*x4+betas1[5]*x5+betas1[6]*x6+
                      betas1[7]*x7+betas1[8]*x8+betas1[9]*x9+
                      betas1[10]*x10+betas1[11]*x11+betas1[12]*x12
                    +betas1Q[1]*x13+ betas1Q[2]*x14+ betas1Q[3]*x15+
                      betas1Q[4]*x16+ betas1Q[5]*x17+ betas1Q[6]*x18)

**Comment by ishwaran** [comment]
I've created some R-code for you to test simulations similar to the 2014 paper.  Here's the link:

https://ishwaran.org/programs/test_CR_simulations.1.1.R

**Reply by Justinewits** [reply]
Thank you very much for your reply, I do appreciate your help. One quick observation in this code is that the values of q.beta1 and 
q.beta2 are not the first three values of beta1 and beta2, respectively as it was in the original paper. According to the original paper, the six  quadratic and the first six linear predictors have the same effect size. This is giving me inconsistent results in my simulation and I am now stuck and only considering using (an illustration based on your example): 

timecause1 ~ quad.Z1+quad.Z2+quad.Z3 
timecause2 ~ quad.Z1+quad.Z2+quad.Z3

Infact, this is my code below on each of the cause specific hazard:
N=200
 x1 <-rnorm(N, mean=0, sd=1) 
  x2 <-rnorm(N, mean=0, sd=1)
  x3 <-rnorm(N, mean=0, sd=1)
  x4 <-rnorm(N, mean=0, sd=1)
  x5 <-rnorm(N, mean=0, sd=1)
  x6 <-rnorm(N, mean=0, sd=1)
  x7 <- rbinom(N,1,0.5)
  x8 <- rbinom(N,1,0.5)
  x9 <- rbinom(N,1,0.5)
  x10 <- rbinom(N,1,0.5)
  x11 <- rbinom(N,1,0.5)
  x12 <- rbinom(N,1,0.5)
  rate = 0.01
  b1 = log(2)
  betas1Q<-c(b1,-b1,0,0,b1,-b1)
  betas2Q<-c(0,0,b1,-b1,b1,-b1)

lambda1<-rate*exp(
                      betas1Q[1]*x1^2+ betas1Q[2]*x2^2+ betas1Q[3]*x3^2+
                      betas1Q[4]*x4^2+ betas1Q[5]*x5^2+ betas1Q[6]*x6^2)
  
  lambda2<-rate*exp(betas2Q[1]*x1^2+
                      betas2Q[2]*x2^2+ betas2Q[3]*x3^2+
                      betas2Q[4]*x4^2+ betas2Q[5]*x5^2+ betas2Q[6]*x6^2)
  # create the overall hazard

**Reply by Justinewits** [reply]
Sorry I forgot to add betas1 and betas2 for comparison:

betas1 <- c(b1,-b1,0,0,b1,-b1,b2,-b2,0,0,b2,-b2)
betas2 <- c(0,0,b1,-b1,b1,-b1,0,0,b2,-b2,b2,-b2)

**Comment by ishwaran** [comment]
Yes, you'll have to modify the code.  But hopefully it will get you started.  Good luck.

**Reply by awhata01** [reply]
Hi Prof
I have a similar question as the one asked above.  

I am trying to perform a similar study to the one in your paper “Random Survival forests for competing risks”. I would like to replicate the linear model, quadratic model, and Interaction model from your paper. I am but confused as to how you formulated the quadratic equation using the predictors x1-x12. 

For the quadratic equation:
1)	Do I simply square the predictors x1-x12 and use the squared variables as the predictors or?
2)	Do I add to the linear model the squared variables x1-x6?

I would be grateful if you could provide guidance on how I can formulate the quadratic models 

Thank you 
Albert Whata
